Actually throw an error if we cannot find min/max version.

EnTeQuAk opened the pull request:
This refs #2970 as the manifest.json specified a wrong
strict_min_version, "45.0.0" which doesn't exist.

This will obviously fail loudly if, say we don't have the same version numbers for Android and Firefox anymore and someone wants to target his WebExtension specifically for that version for the specific app. But, maybe that's just better this way.

maybe cc'ing @wagnerand on this so that more people know this behavior.

diox commented on src/olympia/files/utils.py:
```diff
@@ -314,7 +316,16 @@ def apps(self):
+            msg = _(
+                'Cannot find min/max version. Maybe '
+                '"strict_min_version" or "strict_max_version" '
+                'contains a not supported version?')
```
Wording seems a bit unpleasant to read. Replace the end with `an unsupported version` maybe ?

EnTeQuAk commented:
Alrighty, I fixed up the wording and fixed a few other inconsistencies. I think we cover now all possible scenarios that I can think of:

 * skip the app (e.g android) if the specified `strict_min_version` doesn't support webextensions yet
 * if `strict_min_version` allows webextension support, check if it's in our database. If not, throw an error
 * otherwise just use the `strict_min_version` if it's valid or default to 48.* for Android and 42.0 for Firefox if none is given

Does that make sense to everyone?

diox commented on src/olympia/files/tests/test_utils_.py:
```diff
@@ -244,22 +244,37 @@ def test_summary(self):
             'applications': {
                 'gecko': {
-                    'strict_min_version': '>=30.0',
-                    'strict_max_version': '=30.*',
```
Why change this test ? I kinda like that it's using `>=`/`=`

EnTeQuAk replied:
I re-added those, I just adapted the version numbers to represent something more reasonable since we only support WebExtensions in more recent firefox versions and this tests only tests that the version numbers are accepted. We now fail if the `strict_min_version` is below the min version for webextensions.

diox commented on src/olympia/files/utils.py:
```diff
@@ -308,13 +310,30 @@ def apps(self):
+            )
+
+            if skip_app:
+                continue
```
Add a comment

diox commented on src/olympia/files/utils.py:
```diff
@@ -298,6 +298,8 @@ def apps(self):
                 _('GUID is required for Firefox 47 and below.')
             )
 
+        couldnt_find_version = False
```
Maybe move this down to the next chunk of code that is using it ?

EnTeQuAk replied:
Well it has to be defined before the for-loop since I want it to be set whenever we find an app that does not exist.

diox commented:
r+
